sn_faucet/src/faucet_server.rs

```diff
@@ -202,13 +216,112 @@ fn parse_snapshot(json_str: String) -> Result<Vec<MaidBalance>> {
         // The reserved amount is the amount currently for sale on omni dex.
         // If it's not included the total is lower than expected.
         let r_int = b.reserved.parse::<u32>()?;
-        total += b_int + r_int;
+        let address_balance = b_int + r_int;
+        total += address_balance;
+        balances_map.insert(b.address.clone(), address_balance);
     }
     if total != supply {
         let msg = format!("Incorrect snapshot total, got {total} want {supply}");
         return Err(eyre!(msg));
     }
     // log the total number of balances that were parsed
     info!("Parsed {} maid balances from the snapshot", balances.len());
-    Ok(balances)
+    Ok(balances_map)
+}
+
+fn load_maid_pubkeys() -> Result<HashMap<String, String>> {
```

Comment:
better have a comment explains what are these `String, String` for

Reply:
Changed it to `HashMap<MaidAddress, MaidPubkey>`